File: 250321/떨어지는 1자 블록/falling-horizontal-block.cpp
#include <iostream>

using namespace std;

int n, m, k;
int grid[101][101];

int main() {
    cin >> n >> m >> k;

    for (int i = 1; i <= n; i++) {
        for (int j = 1; j <= n; j++) {
            cin >> grid[i][j];
        }
    }
    
    int row = 0;
    for (int i = 1; i <= n; i++){

        bool is_empty = true;

        for (int j = k; j < k + m; j++) {
            if (grid[i][j] == 1) {
                is_empty = false;
                break;
            }
        }

        if(is_empty) row = i;
    } 
    if(row) {
        for (int j = k; j < k + m; j++) {
            grid[row][j] = 1;
        }
    }
    for (int i = 1; i <= n; i++) {
        for (int j = 1; j <= n; j++) {
            cout << grid[i][j] << ' ';
        }
        cout << '\n';
    }
    return 0;
}

File: 250731/떨어지는 1자 블록/falling-horizontal-block.py
n, m, k = map(int, input().split())
grid = [list(map(int, input().split())) for _ in range(n)]

# 0-based 인덱스 통일
k -= 1

def in_range(x, y):
    return 0 <= x < n and 0 <= y < n

def can_place(row, col):
    for i in range(m):
        if not in_range(row, col + i) or grid[row][col + i] == 1:
            return False
    return True

def simulation():
    row = 0
    while row + 1 < n and can_place(row + 1, k):
        row += 1

    # 최종 위치 row에 블록 m칸 놓기
    for i in range(m):
        grid[row][k + i] = 1

simulation()

for row in grid:
    print(*row)
